### ISSUE TITLE
Universal Connection URL format?

### ISSUE THREAD

**Issue by aguynamedryan** [outcomesinsights/sequelizer#11]
Connection URLs vary quite a bit between different database drivers.

For instance, the pg driver allows a connection string like `postgres://username:password@somehost.example.com/somedatabase` whereas the JDBC driver only allows `jdbc:postgresql://somehost.example.com/somedatabase?username=username&password=password`

In Impala, two of the JDBC drivers have vastly different connection strings when employing Kerberos.  jdbc/impala likes the following:

`jdbc:impala://worker1.hadoop.jsaw.io:21050/default;AuthMech=1;KrbServiceName=impala;KrbAuthType=2;KrbHostFQDN=ip-172-31-15-50.us-west-2.compute.internal;KrbRealm=CLOUDERADEV.OUTINS.COM`

whereas jdbc/hive2 likes:

`jdbc:hive2://worker1.hadoop.jsaw.io:21050/default;principal=impala/ip-172-31-15-50.us-west-2.compute.internal@CLOUDERADEV.OUTINS.COM`

I'd like to find a way to generate the correct connection URL based on a set of options passed into Sequelizer.

@jeremyevans proposed a simple case statement that switches on the value of options[:adapter].  I think this is a good place to start.

```ruby
url = case opts[:adapter]
 when :jdbc_hive2
   # ...
 when :jdbc_impala
   # ...
 when :impala
   # ...
 end
```

From what I can tell, the following options need to be supported:

- username
- password
- realm
- host*
- host_fqdn (defaults to host)
- database
- port

I'd like to focus on support for URLs for:

- PostgreSQL
  - drivers I'd like to support:
     - pg
     - jdbc:postgres
  - username/password is the only thing required.  No need for handling Kerberos.
- Impala
  - drivers I'd like to support:
    - jdbc:hive2
    - jdbc:impala 
    - impala-ruby
  - Handle no authentication and authentication via Kerberos

**Comment by aguynamedryan** [comment]
Also, our existing `config/sequelizer.yml` file looks like this:

```YAML
---
:transport: :<%= ENV["KEYTAB_FILE_PATH"] ? :sasl : :buffered %>
:url: <%= ENV["CLAIMS_RDBMS_URL"] %>
<% if ENV["KEYTAB_FILE_PATH"] %>
  :sasl_params  :mechanism: GSSAPI
  :remote_host: <%= ENV["CLAIMS_RDBMS_REMOTE_HOST"] %>
  :remote_principal: impala
<% end %>
```

I'm not sure which, if any, of those parameters will also need to be included depending on the driver being used.  These parameters seemed to be necessary when using impala-ruby

**Comment by aguynamedryan** [comment]
Closed by #12
